tests/handlers/test_queries.py

```diff
@@ -158,3 +158,30 @@ def test_refresh_query_you_dont_have_access_to(self):
         user = self.factory.create_user(group_ids=[group.id])
         response = self.make_request('post', self.path, user=user)
         self.assertEqual(403, response.status_code)
+
```

Comment:
You could also add tests for QueryVersionListResource, for the fork of a query, and for a revert if you wanted.